components/Recorder.tsx

```diff
@@ -167,23 +169,28 @@ const Recorder = ({ userId }: { userId: string }) => {
     }
   };
 
-  const renderStartStopRecordingButton = () => (
-    <button
-      onClick={handleRecordClick}
-      className={`flex items-center px-6 py-4 mx-auto border border-gray-300 hover:border-orange-500 rounded-full`}
-    >
-      <span
-        className={`block rounded-full bg-${isRunning ? 'error-50 bg-orange-500' : 'orange-500 border-white'} mr-2 border-[1px]`}
+  const renderStartStopRecordingButton = () => {
+    const disabled = isRunning && !mediaRecorder;
+
+    return (
+      <button
+        onClick={handleRecordClick}
+        className={`flex items-center px-6 py-4 mx-auto border border-gray-300 rounded-full ${!disabled && 'hover:border-orange-500'}`}
+        disabled={disabled}
       >
         <span
-          className={`block w-3 h-3 m-3 ${!isRunning && 'rounded-full'} bg-white`}
-        ></span>
-      </span>
-      <span className="text-2xl">
-        {isRunning ? 'Stop recording' : 'Record a note'}
-      </span>
-    </button>
-  );
+          className={`block rounded-full bg-${isRunning ? 'error-50 bg-orange-500' : 'orange-500 border-white'} mr-2 border-[1px]`}
```

Comment:
This will not work with tailwind JIT compiler
https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Reply:
Done!